runtime/lua/vim/treesitter/query.lua

```diff
@@ -799,6 +962,18 @@ function Query:iter_matches(node, source, start, stop, opts)
 
       self:apply_directives(match, pattern, source, metadata)
     end
+
+    if not opts or not opts.correct then
```

Comment:
Not sure we need  `vim.deprecate` actually. The warning wouldn't appear until the 0.11 release cycle, but in that release cycle the option is being changed to be the default anyway (per @clason's [accelerated timeline](https://github.com/neovim/neovim/pull/24738#issuecomment-1943318653), which I agree with since treesitter still has experimental status).